sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/JavaTypeInference.scala

```diff
@@ -309,7 +307,7 @@ object JavaTypeInference {
             dataType,
             nullable = elementNullable,
             newTypePath,
-            (casted, typePath) => deserializerFor(et, casted, typePath))
+            deserializerFor(et, _, newTypePath))
```

Comment:
I think the changes in this file are just code cleanup. I'm fine with it as they are also related.